app/javascript/components/star-rating.vue

```diff
@@ -0,0 +1,46 @@
+
+<script setup lang="ts">
+
+type Props = {
+  rating: number | string
+}
+
+const MAX_STARS = 5;
+
+const props = defineProps<Props>();
+
+function halfStars() {
+  return props.rating % 1 !== 0;
```

Comment:
🚫 **[tsc]** <2362> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.